addons/mail/static/src/discuss/core/common/notification_settings.js

```diff
@@ -0,0 +1,97 @@
+/* @odoo-module */
+
+import { useState, Component } from "@odoo/owl";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+
+export class NotificationSettings extends Component {
+    static components = { Dropdown, DropdownItem };
+    static props = ["hasSizeConstraints?", "thread", "close", "className?"];
+    static template = "discuss.NotificationSettings";
+
+    setup() {
+        this.threadService = useState(useService("mail.thread"));
+    }
+
+    get settingItems() {
+        return [
+            {
+                id: "all",
+                name: _t("All Messages"),
+            },
+            {
+                id: "mentions",
+                name: _t("Mentions Only"),
+            },
+            {
+                id: "none",
+                name: _t("Nothing"),
+            },
+        ];
+    }
```

Comment:
At a glance the ids look arbitrary, but actually they need to match `thread.notification_type` values.
I would move this enum in `Thread` model, so that the locality of concern is better.